### PR TITLE
Remove method router

### PR DESCRIPTION
**Pull request by syllerim**

### CODE REVIEW

File: ApiConnector/NetworkConnector.swift [hunk 1]
```diff
@@ -26,20 +26,21 @@ public protocol ApiConnectionType {
     var defaultHeaders: HTTPHeaders? { get }
     var defaultValidation: Alamofire.DataRequest.Validation? { get }
     
-    func request(with data: Data?, at endpoint: RouterType, headers: HTTPHeaders?) -> URLRequest
-    func requestData(with data: Data?, at endpoint: RouterType, headers: HTTPHeaders?) -> RequestType
+    func request(method: HTTPMethod?, with data: Data?, at endpoint: RouterType, headers: HTTPHeaders?) -> URLRequest
```
Comment (alickbass) [inline ApiConnector/NetworkConnector.swift]: To me it doesn't make sense why method is `Optional`. If you want to add default value for the parameter, do it like this:

```swift
func request(method: HTTPMethod = .get, with data: Data?, at endpoint: RouterType, headers: HTTPHeaders?) -> URLRequest
```

File: ApiConnector/JSONModelNetworkConnector.swift [hunk 2]
```diff
@@ -30,27 +30,28 @@ public extension DataRequestType {
 }
 
 public extension ApiConnectionType {
-    public func requestData(with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> RequestType {
-        return requestData(with: model?.jsonValue, at: endpoint, headers: headers)
+    
+    public func requestData(method: HTTPMethod, with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> RequestType {
```
Comment (alickbass) [inline ApiConnector/JSONModelNetworkConnector.swift]: Can you add default method to be `get` here?

File: ApiConnector/JSONModelNetworkConnector.swift [hunk 3]
```diff
@@ -30,27 +30,28 @@ public extension DataRequestType {
...
+        return requestData(method: method, with: model?.jsonValue, at: endpoint, headers: headers)
     }
     
-    public func requestData(with json: JSON? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> RequestType {
-        return requestData(with: json.flatMap({ try? $0.rawData() }), at: endpoint, headers: headers)
+    public func requestData(method: HTTPMethod, with json: JSON? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> RequestType {
```
Comment (alickbass) [inline ApiConnector/JSONModelNetworkConnector.swift]: And here

File: ApiConnector/JSONModelNetworkConnector.swift [hunk 4]
```diff
@@ -30,27 +30,28 @@ public extension DataRequestType {
...
+        return requestData(method: method, with: json.flatMap({ try? $0.rawData() }), at: endpoint, headers: headers)
     }
     
-    public func requestObservable<T: JSONInitializable>(with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<T> {
-        return validate(request: requestData(with: model, at: endpoint, headers: headers)).modelObservable()
+    public func requestObservable<T: JSONInitializable>(method: HTTPMethod, with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<T> {
```
Comment (alickbass) [inline ApiConnector/JSONModelNetworkConnector.swift]: And here

File: ApiConnector/JSONModelNetworkConnector.swift [hunk 5]
```diff
@@ -30,27 +30,28 @@ public extension DataRequestType {
...
+        return validate(request: requestData(method: method, with: model, at: endpoint, headers: headers)).modelObservable()
     }
     
-    public func requestObservable<T: JSONInitializable>(with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<[T]> {
-        return validate(request: requestData(with: model, at: endpoint, headers: headers)).modelObservable()
+    public func requestObservable<T: JSONInitializable>(method: HTTPMethod, with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<[T]> {
```
Comment (alickbass) [inline ApiConnector/JSONModelNetworkConnector.swift]: and here

File: ApiConnector/JSONModelNetworkConnector.swift [hunk 6]
```diff
@@ -30,27 +30,28 @@ public extension DataRequestType {
...
+        return validate(request: requestData(method: method, with: model, at: endpoint, headers: headers)).modelObservable()
     }
     
-    public func requestObservable(with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<Void> {
-        return validate(request: requestData(with: model, at: endpoint, headers: headers)).observable()
+    public func requestObservable(method: HTTPMethod, with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<Void> {
```
Comment (alickbass) [inline ApiConnector/JSONModelNetworkConnector.swift]: And here

File: ApiConnector/JSONModelNetworkConnector.swift [hunk 7]
```diff
@@ -30,27 +30,28 @@ public extension DataRequestType {
...
+        return validate(request: requestData(method: method, with: model, at: endpoint, headers: headers)).observable()
     }
     
-    public func requestObservable(with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<JSON> {
-        return validate(request: requestData(with: model, at: endpoint, headers: headers)).jsonObservable()
+    public func requestObservable(method: HTTPMethod, with model: JSONRepresentable? = nil, at endpoint: RouterType, headers: HTTPHeaders? = nil) -> Observable<JSON> {
```
Comment (alickbass) [inline ApiConnector/JSONModelNetworkConnector.swift]: And here

File: ApiConnectorTests/JSONModelNetworkConnectorTests.swift [hunk 8]
```diff
@@ -23,7 +23,7 @@ class JSONModelNetworkConnectorTests: XCTestCase {
     func testModelRequest() {
         let postExpactation = expectation(description: "ModelRequestExpectation")
         let post = TestData.defaultPost
-        let postObservable: Observable<Post> = Connection(environment: .test).requestObservable(with: post, at: .pictures)
+        let postObservable: Observable<Post> = Connection(environment: .test).requestObservable(method: .get, with: post, at: .pictures)
```
Comment (alickbass) [inline ApiConnectorTests/JSONModelNetworkConnectorTests.swift]: When you add default method param, all these changes should be discarded, as they won't be necessary

**Review by alickbass — CHANGES_REQUESTED** [review]

File: ApiConnectorTests/ApiConnectionTests.swift [hunk 9]
```diff
@@ -40,7 +40,7 @@ class ApiConnectionTests: XCTestCase {
     func testcustomRequestValidation() {
         let connector = TestValidationConnection<SuccessProvider>()
         let responseExpectation = expectation(description: "SuccessMockResponse")
-        let request = connector.requestData(with: nil as Data?, at: .me, headers: nil)
+        let request = connector.requestData(method: .get,with: nil as Data?, at: .me, headers: nil)
```
Comment (alickbass) [inline ApiConnectorTests/ApiConnectionTests.swift]: These changes can be removed

File: ApiConnector/NetworkConnector.swift (hunk 1 above)
Comment (syllerim) [reply ApiConnector/NetworkConnector.swift]: Yep, make sense.

File: ApiConnector/JSONModelNetworkConnector.swift (hunk 2 above)
Comment (syllerim) [reply ApiConnector/JSONModelNetworkConnector.swift]: done

File: ApiConnector/JSONModelNetworkConnector.swift (hunk 3 above)
Comment (syllerim) [reply ApiConnector/JSONModelNetworkConnector.swift]: done

File: ApiConnector/JSONModelNetworkConnector.swift (hunk 4 above)
Comment (syllerim) [reply ApiConnector/JSONModelNetworkConnector.swift]: done

File: ApiConnectorTests/RxNetworkConnectorTests.swift [hunk 10]
```diff
@@ -13,7 +13,7 @@ class RxNetworkConnectorTests: XCTestCase {
     
     func testApiObservableSuccess() {
         let successExpectation = expectation(description: "ApiObservableExpectation")
-        let observable = TestApiConnection<SuccessProvider>(environment: .test).requestData(with: nil as Data?, at: .me, headers: nil).responseObservable().subscribe(onNext: { data in
+        let observable = TestApiConnection<SuccessProvider>(environment: .test).requestData(method: .get, with: nil as Data?, at: .me, headers: nil).responseObservable().subscribe(onNext: { data in
```
Comment (alickbass) [inline ApiConnectorTests/RxNetworkConnectorTests.swift]: These changes are also not necessary

File: ApiConnector/JSONModelNetworkConnector.swift (hunk 5 above)
Comment (syllerim) [reply ApiConnector/JSONModelNetworkConnector.swift]: done

File: ApiConnector/JSONModelNetworkConnector.swift (hunk 6 above)
Comment (syllerim) [reply ApiConnector/JSONModelNetworkConnector.swift]: done

File: ApiConnector/JSONModelNetworkConnector.swift (hunk 7 above)
Comment (syllerim) [reply ApiConnector/JSONModelNetworkConnector.swift]: done

File: ApiConnectorTests/JSONModelNetworkConnectorTests.swift (hunk 8 above)
Comment (syllerim) [reply ApiConnectorTests/JSONModelNetworkConnectorTests.swift]: yep, make sense

File: ApiConnectorTests/RxNetworkConnectorTests.swift [hunk 11]
```diff
@@ -24,7 +24,7 @@ class RxNetworkConnectorTests: XCTestCase {
     
     func testApiObservableFailure() {
         let successExpectation = expectation(description: "ApiObservableExpectation")
-        let observable = TestApiConnection<SuccessProvider>(environment: .test).requestData(with: nil as Data?, at: .me, headers: nil).validate({ _,_,_ in
+        let observable = TestApiConnection<SuccessProvider>(environment: .test).requestData(method: .get, with: nil as Data?, at: .me, headers: nil).validate({ _,_,_ in
```
Comment (alickbass) [inline ApiConnectorTests/RxNetworkConnectorTests.swift]: These changes are also not necessary

**Review by alickbass — CHANGES_REQUESTED** [review]

File: ApiConnectorTests/ApiConnectionTests.swift (hunk 9 above)
Comment (syllerim) [reply ApiConnectorTests/ApiConnectionTests.swift]: removed

File: ApiConnectorTests/RxNetworkConnectorTests.swift (hunk 10 above)
Comment (syllerim) [reply ApiConnectorTests/RxNetworkConnectorTests.swift]: removed

File: ApiConnectorTests/RxNetworkConnectorTests.swift (hunk 11 above)
Comment (syllerim) [reply ApiConnectorTests/RxNetworkConnectorTests.swift]: removed

**Review by alickbass — APPROVED** [review]

**Pull request merged**